### PR TITLE
chore: use @bigcommerce/eslint-config instead of custom config

### PR DESCRIPTION
**Pull request by chanceaclark**
## What?

This PR will consume Big Commerce's public eslint config. **We are going to be disabling all the manual fixes for now**, and create new PRs to address them on a more "1 by 1" basis (we may tackle 1 or more in a single PR).

For reviewing, reviewing by commits will be helpful:
1. Update eslint config https://github.com/bigcommerce/big-design/pull/863/commits/7952c53793e3765bb4a1029efc94d1b813b51ec6
2. `eslint --fix` https://github.com/bigcommerce/big-design/pull/863/commits/7c7ef88578da32a931d58a9386cbe0c0b4eba6b4
3. Disable eslint rules that need manual fixing: https://github.com/bigcommerce/big-design/pull/863/commits/af3878e2a96f40edd8bfe5b9a93421569d4c0f7a

## Why?

This will align big-design's linting standards with Big Commerce's public configuration. This gives us the foundations for consistency across BigCommerce repositories and will provide us with a better coding standards in our component library.

## Screenshots/Screen Recordings

N/A

## Testing/Proof

yarn lint has successfully passed with disabled rules and autofixes

### CODE REVIEW

File: packages/big-design/src/components/Search/spec.tsx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { Search } from './Search';
 test('renders the search component', () => {
   render(<Search onChange={jest.fn()} onSubmit={jest.fn()} value="Product" />);
 
-  const input = screen.getByLabelText('Search');
+  const input = screen.getByLabelText<HTMLInputElement>('Search');
```
Comment (chanceaclark) [inline packages/big-design/src/components/Search/spec.tsx:9]: TypeScript was complaining that `input.value` wasn't a thing.

File: packages/big-design/src/components/StatefulTable/spec.tsx [hunk 2]
```diff
@@ -71,7 +71,7 @@ test('dragAndDrop can be enabled', async () => {
   const onRowDrop = jest.fn();
   const { container, getAllByTestId } = render(getSimpleTable({ onRowDrop }));
 
-  let dragEls = container.querySelectorAll('[data-rbd-draggable-id]');
+  let dragEls = container.querySelectorAll<HTMLButtonElement>('[data-rbd-draggable-id]');
```
Comment (chanceaclark) [inline packages/big-design/src/components/StatefulTable/spec.tsx:74]: TypeScript was complaining that `dragEls[0].focus()` wasn't a thing.

File: packages/big-design/src/managers/alerts/spec.tsx [hunk 3]
```diff
@@ -187,7 +187,7 @@ describe('alertsManager functionality', () => {
   });
 
   test('subscibes to updates', () => {
-    const alerts = [];
+    const alerts: AlertProps[] = [];
```
Comment (chanceaclark) [inline packages/big-design/src/managers/alerts/spec.tsx:190]: TypeScript was complaining that `alerts` was `any[]`.

**Review by bc-juanvasquez — APPROVED** [review]

**Review by deini — APPROVED** [review]

**Review by MariaJose — APPROVED** [review]
👏🏼 !

**Review by rtalvarez — APPROVED** [review]

**Pull request merged**